frontend/src/components/ProvideSqlEditorContext.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <slot />
+</template>
+
+<script lang="ts" setup>
+import { watchEffect } from "vue";
+import { useStore } from "vuex";
+import { Instance, Database } from "../types";
+// import { idFromSlug } from "../utils";
+
+const store = useStore();
+
+const prepareSqlEdtiorContext = async function () {
+  let connectionTree = []
+
+  const instances = await store.dispatch("instance/fetchInstanceList");
+  connectionTree = instances.map((item: Instance) => ({
```

Comment:
So, if we are assigning a list of instance to it, then I would just call it instanceList.

And because we have a defined tree structure instance -> database -> table, I would prefer to have a dedicate type for each model instead of using a generic ConnectionTree type.